nixos/modules/hardware/video/nvidia.nix

```diff
@@ -274,9 +274,15 @@ in
       Option "AllowNVIDIAGPUScreens"
     '';
 
-    services.xserver.displayManager.setupCommands = optionalString syncCfg.enable ''
+    services.xserver.displayManager.setupCommands = let
+      sinkGpuProviderName = if igpuDriver == "amdgpu" then
+        # find the name of the provider if amdgpu
+        "`${pkgs.xorg.xrandr}/bin/xrandr --listproviders | ${pkgs.gnugrep}/bin/grep -i AMD | ${pkgs.gnused}/bin/sed -n 's/^.*name://p'`"
```

Comment:
Unfortunately AMD's provider naming is really poor.  On my system it is `Unknown AMD Radeon GPU @ pci:0000:04:00.0` I have read that this name has changed format several times and there isn't a way to set the name afaik.  I am fairly confident that it will have "amd" somewhere in it even if amd changes the naming scheme though.

Reply:
On a ASUSTeK COMPUTER INC. ROG Zephyrus G15 GA503QR_GA503QR the `xrandr --listproviders` output is like:

```
Providers: number : 2
Provider 0: id: 0x45 cap: 0x0 crtcs: 4 outputs: 2 associated providers: 0 name:Unknown AMD Radeon GPU @ pci:0000:06:00.0
Provider 1: id: 0x1eb cap: 0x2, Sink Output crtcs: 4 outputs: 4 associated providers: 0 name:NVIDIA-G0
```

The expression here outputs `Unknown AMD Radeon GPU @ pci:0000:06:00.0` which seems to be the intended match to me.

It might not hurt to include such example output near this expression to make it clear what the text looked like when the expression was written (and so make it clear to future maintainers what text it is *trying* to match against).

For what it's worth (probably not much), `lspci` manages to discover this string:

```
VGA compatible controller: Advanced Micro Devices, Inc. [AMD/ATI] Cezanne (rev c4) (prog-if 00 [VGA controller])
```

and `hwinfo` figures out all of:

```
  Model: "ATI Cezanne"
  Vendor: pci 0x1002 "ATI Technologies Inc"
  Device: pci 0x1638 "Cezanne"
```

"Cezanne" appears to be the codename for this series of APU.

Separately, as far as I can tell, the way grep and sed executables are found by this expression looks good.